xbmc/addons/AddonInstaller.cpp

```diff
@@ -828,6 +828,8 @@ bool CAddonUnInstallJob::DoWork()
   }
 
   ClearFavourites();
+  if (m_remove)
+    CFileUtils::DeleteItem("special://profile/addon_data/"+m_addon->ID()+"/", true);
```

Comment:
not sure i understand the question. if a user substitute a non-writable path, it will fail silently. if you are asking whether the substitiution will take place, yes, that happens on a much deeper layer in the vfs.